titus-api/src/main/java/com/netflix/titus/api/jobmanager/store/mixin/VolumeMixin.java

```diff
@@ -0,0 +1,21 @@
+package com.netflix.titus.api.jobmanager.store.mixin;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.netflix.titus.api.jobmanager.model.job.SharedContainerVolumeSource;
+
+public abstract class VolumeMixin {
+    @JsonCreator
+    public VolumeMixin(
+            @JsonProperty("name")
+                    String name,
+
+            @JsonInclude(Include.NON_NULL)
```

Comment:
Why is this annotation here?

Reply:
Great question.

My reasoning here is that as we add all sorts of different configs for volumes for new types (ebs, efs, etc):
https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#volume-v1-core

I didn't want to polute the json with all sorts of non-related configs.

So if figured for this case, we only want to serialize configs that are related to the volume in question (ie if they are non-null.